core/chainio/avs_reader.go

```diff
@@ -54,3 +60,31 @@ func (r *AvsReader) GetErc20Mock(tokenAddr gethcommon.Address) (*contractERC20Mo
 	}
 	return erc20Mock, nil
 }
+
+func (r *AvsReader) GetNewTaskCreated(taskIndex uint64) (*contractAlignedLayerServiceManager.ContractAlignedLayerServiceManagerNewTaskCreated, error) {
+	latestBlock, err := r.AvsContractBindings.ethClient.BlockNumber(context.Background())
+	if err != nil {
+		r.logger.Error("Failed to get latest block number", "err", err)
+		return nil, err
+	}
+	startBlock := uint64(0)
+	if latestBlock > blockRange {
+		startBlock = latestBlock - blockRange
+	}
+	filterOpts := bind.FilterOpts{
+		Start: startBlock,
+	}
+
+	itr, err := r.AvsContractBindings.ServiceManager.FilterNewTaskCreated(&filterOpts, []uint64{taskIndex})
+	if err != nil {
+		return nil, err
+	}
+
+	for itr.Next() {
+		event := itr.Event
+		if event.TaskIndex == taskIndex {
+			return event, nil
+		}
+	}
```

Comment:
Could be replaced with one call to itr.Next, since we only query about one task index

Reply:
Done